pkg/settings/setting.go

```diff
@@ -235,3 +235,13 @@ func getMetadataConfig() string {
 	}
 	return string(ans)
 }
+
+// getSettingByID returns a setting that is stored with the given id. If none is found
+// an error is return
+func GetSettingByID(id string) (Setting, error) {
+	setting, ok := settings[id]
+	if !ok {
+		return Setting{}, fmt.Errorf("%s is not a valid setting", id)
+	}
+	return setting, nil
+}
```

Comment:
This needs to use the provider, if there is one, otherwise you will be reading the value as it lives in this file, not using values set through the env OR in the k8s setting CRD.

Reply:
That is as simple as returning `setting.Get()` right ?